### DISCUSSION TITLE
Get IStorageFolder for FilePickerSaveOptions.SuggestedStartLocation

### DISCUSSION THREAD

**Discussion by CYLMos** [AvaloniaUI/Avalonia#9953] [Q&A]
Now I want to save a  file and open a save dialog to save it.

And I want to specify the default folder for this save dialog but I have no idea to create a `IStorageFolder` object and assign it to `SuggestedStartLocation`

Thanks!

**Comment by maxkatz6** [comment]
In current builds you need to ask user to pick the folder first. You also can save previously picked folder to bookmarks and serialize it.
But it is planned to add couple of additional helper methods. Primarily to be able to open well known folders (pictures, documents, downloads...) without asking it first.

**Reply by CYLMos** [reply]
@maxkatz6 "In current builds you need to ask user to pick the folder first." <-- I don't understand this part.
If the folder location is decided by me, not user, is it possible that I use a absolute path to get `IStorageFolder`?

Like the obsolete method:
`var dialog = new SaveFileDialog { InitialFileName = "File.json", Directory = "My Path", DefaultExtension = "json" };`

**Reply by AscendLiu** [reply]
Hello, how did you solve it

**Reply by maxkatz6** [reply]
@AscendLiu 
`StorageProvider` has `TryGetFolderFromPathAsync` method now. You can use it.
But even better you can use `TryGetWellKnownFolderAsync` method, if you need one of pre-defined user folders, like `Downloads`.

**Reply by AscendLiu** [reply]
I don't quite understand why it's recommended to use `TryGetWellKnownFolderAsync`, even though it's more common to use opening a specified path to select files and often requires attaching `FilePickerOpenOptions`. Finally I use the `OpenFilePickerAsync`
